nixos/modules/services/x11/compton.nix

```diff
@@ -189,15 +194,11 @@ in {
     };
 
     vSync = mkOption {
-      type = types.enum [
```

Comment:
To maintain backwards compatibility I'd suggest having

    with types; either bool (enum [ "none" … "opengl-mswc" ])

instead and then adding something like
```
apply = x:
  let
    res = x != "none";
    msg =
      "The type of services.compton.vSync has changed to bool:"
      + " interpreting ${x} as ${res}";
  in
    if isBool x then x
    else warn msg res;
```

Reply:
Alright, did that as a separate commit so that we can revert it easily in the future when it's reasonable to assume everyone has upgraded.

Reply:
Nice! But I was thinking of using the actual [`apply` function](https://github.com/NixOS/nixpkgs/blob/6d3b7390bc683eea884f9e1d5846632e4fe10428/lib/options.nix#L43-L44) of the `mkOption` argument.

See, for example, in `system-environment.nix`: https://github.com/NixOS/nixpkgs/blob/6d3b7390bc683eea884f9e1d5846632e4fe10428/nixos/modules/config/system-environment.nix#L17-L28

Reply:
Ohh, thanks! I didn't know that